node/legacy_wrapper.go

```diff
@@ -0,0 +1,79 @@
+package node
+
+import (
+	"context"
+	"time"
+
+	pkgerrors "github.com/pkg/errors"
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	"github.com/virtual-kubelet/virtual-kubelet/trace"
+	v1 "k8s.io/api/core/v1"
+)
+
+type legacyPodLifecycleHandlerWrapper struct {
+	PodLifecycleHandlerV0
+	notifier      func(*v1.Pod)
+	reconcileTime time.Duration
+}
+
+func (wrapper *legacyPodLifecycleHandlerWrapper) NotifyPods(ctx context.Context, notifier func(*v1.Pod)) {
+	wrapper.notifier = notifier
+}
+
+// updatePodStatuses syncs the providers pod status with the kubernetes pod status.
+func (wrapper *legacyPodLifecycleHandlerWrapper) updatePodStatuses(ctx context.Context) {
+	ctx, span := trace.StartSpan(ctx, "updatePodStatuses")
+	defer span.End()
+
+	pods, err := wrapper.GetPods(ctx)
+	if err != nil {
+		err = pkgerrors.Wrap(err, "error getting pod list")
+		span.SetStatus(err)
+		log.G(ctx).WithError(err).Error("Error updating pod statuses")
+		return
+	}
+
+	for _, pod := range pods {
+		// Notifier is idempotent.
+		wrapper.notifier(pod)
+	}
+}
+
+func (wrapper *legacyPodLifecycleHandlerWrapper) run(ctx context.Context) {
+	for {
+		t := time.NewTimer(wrapper.reconcileTime)
+		defer t.Stop()
+
+		for {
```

Comment:
What's the purpose of this double loop?

Reply:
Errr, this was from a change before I made run() a private function.